### PR TITLE
New extension: LIFX 1.2

### PR DESCRIPTION
**Pull request by felixb1010**
## Description

This extensions enables you to quickly and easily control your LIFX lights by controlling individual lights, or by setting scenes.

## Screencast

[Screencast](https://www.loom.com/share/382c191296d94788a6a2e6086df0ade6)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/lifx/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+  async function fetchLights() {
+    try {
+      if (!cache.has("lights")) {
+        const isTokenValid = await checkApiKey();
+        if (!isTokenValid) {
+          cache.set("lifx_token", JSON.stringify({ valid: true }));
+          await showToast({
+            style: Toast.Style.Failure,
+            title: "Invalid Token",
```
Comment (pernielsentikaer) [inline extensions/lifx/src/index.tsx:33]:
```suggestion
            title: "Invalid Token",
            message: "Please check your token and try again",
```
Maybe something like this for error toasts?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @felixb1010 👋 

Thanks for your contribution 🔥 

I have some feedback for you to address:

- It could be awesome if all the views could have the same toast and List.Emptyview like this one - having the extension logo in 64x64px

image

- Is there a way to check the token with the `Set Effect` command too, right now I'm just entering the list and can see a `Move` item which I'm not sure what it actually does.

File: extensions/lifx/src/index.tsx (hunk 1 above)
Comment (felixb1010) [reply extensions/lifx/src/index.tsx:33]: Yeah, I am going to add check token to all the functions interacting with the LIFX API and the set effect command.

**Comment by felixb1010** [comment]
> Hi @felixb1010 👋
> 
> Thanks for your contribution 🔥
> 
> I have some feedback for you to address:
> 
> * It could be awesome if all the views could have the same toast and List.Emptyview like this one - having the extension logo in 64x64px
> 
> image
> 
> * Is there a way to check the token with the `Set Effect` command too, right now I'm just entering the list and can see a `Move` item which I'm not sure what it actually does.

The move item is a light effect. Should I make it more clear by adding a subtitle?

**Comment by pernielsentikaer** [comment]
Is there no way to get `Set Effect` to return the same error as the other commands initially @felixb1010

**Comment by felixb1010** [comment]
Like using check token before i show the effects

**Comment by pernielsentikaer** [comment]
Yes, so we get the same UX all the way around 🙂

**Comment by felixb1010** [comment]
Should I add a toast for fetching light when it is using the cache or only for when its fetching from the api

**Comment by pernielsentikaer** [comment]
I'm not that much into the extension, if it's require a lot of work to get the last command to do this, then we shouldn't do it now :)

File: extensions/lifx/package.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      "mode": "no-view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "lifx_token",
+      "title": "LIFX Token",
+      "description": "Your LIFX token",
+      "type": "textfield",
+      "default": "Token",
```
Comment (pernielsentikaer) [inline extensions/lifx/package.json:58]:
```suggestion
      "placeholder": "xfergfrtetrektopfksdopfewr",
```
...Or something like that 😃

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
It seems like there still are a problem with `Set Effect`, here is just keeps spinning without anything happening (I don't have a token)

CleanShot 2022-09-12 at 14 39 10@2x

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**